src/main/java/com/semmle/jira/addon/Request.java

```diff
@@ -182,34 +183,52 @@ boolean isValid() {
 
     @JsonIgnoreProperties(ignoreUnknown = true)
     public static class Query {
-      @JsonProperty public final String name;
+      @Deprecated @JsonProperty private final String name;
```

Comment:
As the name is mandatory for the query to be valid I would keep it here instead of adding it to `properties`. The properties seems like complimentary information that can be or can not be in there.

Reply:
The properties object corresponds to the query metadata specified in the header comment of a .ql file. For example `@name` ,  `@id` and `@tags` .